client/analytics/report/orders/table.js

```diff
@@ -0,0 +1,267 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { format as formatDate } from '@wordpress/date';
+import { map, orderBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { Card, TableCard, TablePlaceholder } from '@woocommerce/components';
+import { downloadCSVFile, generateCSVDataFromTable, generateCSVFileName } from 'lib/csv';
+import { formatCurrency, getCurrencyFormatDecimal } from 'lib/currency';
+import { getIntervalForQuery, getDateFormatsForInterval } from 'lib/date';
+import { getAdminLink, onQueryChange } from 'lib/nav-utils';
+
+export default class OrdersReportTable extends Component {
+	constructor( props ) {
+		super( props );
+	}
+
+	onDownload( headers, rows, query ) {
+		// @TODO The current implementation only downloads the contents displayed in the table.
+		// Another solution is required when the data set is larger (see #311).
+		return () => {
+			downloadCSVFile(
+				generateCSVFileName( 'orders', query ),
+				generateCSVDataFromTable( headers, rows )
+			);
+		};
+	}
+
+	getHeadersContent() {
+		return [
+			{
+				label: __( 'Date', 'wc-admin' ),
+				key: 'date_created',
+				required: true,
+				defaultSort: true,
+				isIdentifier: true,
+				isSortable: true,
+			},
+			{
+				label: __( 'Order #', 'wc-admin' ),
+				key: 'id',
+				required: true,
+				isIdentifier: true,
+				isSortable: true,
+			},
+			{
+				label: __( 'Status', 'wc-admin' ),
+				key: 'status',
+				required: false,
+				isSortable: true,
+			},
+			{
+				label: __( 'Customer', 'wc-admin' ),
+				key: 'customer_id',
+				required: false,
+				isSortable: true,
+			},
+			{
+				label: __( 'Product(s)', 'wc-admin' ),
+				key: 'products',
+				required: false,
+				isSortable: false,
+			},
+			{
+				label: __( 'Items Sold', 'wc-admin' ),
+				key: 'items_sold',
+				required: false,
+				isSortable: true,
+				isNumeric: true,
+			},
+			{
+				label: __( 'Coupon(s)', 'wc-admin' ),
+				key: 'coupons',
+				required: false,
+				isSortable: false,
+			},
+			{
+				label: __( 'N. Revenue', 'wc-admin' ),
+				key: 'net_revenue',
+				required: false,
```

Comment:
This should probably be `true`, otherwise you can turn off all useful columns in the report

Reply:
Right, I was not sure which ones should be required and which shouldn't. Now `date_created`, `id` and `net_revenue` will be required and all others won't. Is that a good approach?

Reply:
yeah, that sounds good – I think that's the minimum needed for the table to be useful